kairon/api/app/routers/bot/data.py

```
@@ -170,6 +171,30 @@ async def delete_cognition_data(
+    """
+    if not row_ids:
+        raise AppException("row_ids list cannot be empty!")  # ✅ Handling empty list case
+
+    try:
+        query = {"id__in": row_ids}
+        fetched_documents = CognitionData.objects(**query)
+        if not fetched_documents:
+            raise AppException("Some or all records do not exist!")
+        Utility.hard_delete_document([CognitionData], bot=current_user.get_bot(), **query, user=current_user.get_user())
+    except DoesNotExist:
+        raise AppException("Some or all records do not exist!")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)